lib/models/device.coffee

```diff
@@ -99,6 +100,15 @@ getDeviceModel = (deps, opts) ->
 	if not dashboardUrl?
 		dashboardUrl = apiUrl.replace(/api/, 'dashboard')
 
+	getDeviceUrlsBase = once ->
+		if deviceUrlsBase?
+			return deviceUrlsBase
```

Comment:
Is this also a Promise? Otherwise I assume it will throw when you call `.then` on it.

Reply:
Oh nice catch @sradevski :+1: 
This was defenitelly going to cause issues if someone used the new constructor parameter!
Will update that shortly :+1:

Reply:
So this is still pending? Or was resolved in an interation?

Reply:
That's resolved @imrehg . Wrapping with `Promise.method` ensures that `getDeviceUrlsBase()` will always return a promise.